Two test failures:
```
_________________ NotificationsAPITestCase.test_create_examlog _________________
self = <kolibri.core.notifications.test.test_api.NotificationsAPITestCase testMethod=test_create_examlog>
save_notifications = <MagicMock name='save_notifications' id='139907087624784'>
    @patch('kolibri.core.notifications.api.save_notifications')
    def test_create_examlog(self, save_notifications):
        examlog = ExamLog.objects.create(exam=self.exam, user=self.user1)
>       create_examlog(examlog)
E       TypeError: create_examlog() takes exactly 2 arguments (1 given)
kolibri/core/notifications/test/test_api.py:199: TypeError
----------------------------- Captured stdout call -----------------------------
Installed 2 object(s) from 1 fixture(s)
_________________ NotificationsAPITestCase.test_parse_examlog __________________
self = <kolibri.core.notifications.test.test_api.NotificationsAPITestCase testMethod=test_parse_examlog>
save_notifications = <MagicMock name='save_notifications' id='139907089515152'>
    @patch('kolibri.core.notifications.api.save_notifications')
    def test_parse_examlog(self, save_notifications):
        examlog = ExamLog.objects.create(exam=self.exam, user=self.user1)
>       parse_examlog(examlog)
E       TypeError: parse_examlog() takes exactly 2 arguments (1 given)
kolibri/core/notifications/test/test_api.py:187: TypeError
----------------------------- Captured stdout call -----------------------------
Installed 2 object(s) from 1 fixture(s)
```

--- a/kolibri/core/notifications/api.py
+++ b/kolibri/core/notifications/api.py
@@ -337,6 +337,6 @@ def parse_attemptslog(attemptlog):
                                                lesson.group_or_classroom,
                                                lesson_id=lesson.id,
                                                contentnode_id=contentnode_id,
-                                               reason=HelpReason.Multiple)
+                                               timestamp=attemptlog.start_timestamp)
             notifications.append(notification)
         save_notifications(notifications)

--- a/kolibri/core/notifications/api.py
+++ b/kolibri/core/notifications/api.py
@@ -320,3 +320,23 @@ def parse_attemptslog(attemptlog):
                                                timestamp=attemptlog.end_timestamp)
             notifications.append(notification)
         save_notifications(notifications)
+
+    else:
+        notifications = []
+        for lesson, contentnode_id in lessons:
+            if LearnerProgressNotification.objects.filter(user_id=attemptlog.user_id,
+                                                          notification_object=NotificationObjectType.Resource,
+                                                          notification_event=NotificationEventType.Started,
+                                                          lesson_id=lesson.id,
+                                                          classroom_id=lesson.group_or_classroom,
+                                                          contentnode_id=contentnode_id).exists():
+                continue
+            notification = create_notification(NotificationObjectType.Resource,
+                                               NotificationEventType.Started,
+                                               attemptlog.user_id,
+                                               lesson.group_or_classroom,
+                                               lesson_id=lesson.id,
+                                               contentnode_id=contentnode_id,
+                                               reason=HelpReason.Multiple)
+            notifications.append(notification)
+        save_notifications(notifications)

--- a/kolibri/core/notifications/api.py
+++ b/kolibri/core/notifications/api.py
@@ -15,11 +15,11 @@ from kolibri.core.logger.models import ContentSummaryLog
 
 
 @memoize
-def get_assignments(instance, summarylog, attempt=False):
+def get_assignments(user, summarylog, attempt=False):
     """
     Returns all Lessons assigned to the user having the content_id
     """
-    memberships = instance.user.memberships.all()
+    memberships = user.memberships.all()
     # If the user is not in any classroom nor group, nothing to notify
     if not memberships:
         return []
@@ -176,7 +176,7 @@ def create_summarylog(summarylog):
     summarylog is created.
     It creates the Resource and, if needed, the Lesson Started event
     """
-    lessons = get_assignments(summarylog, summarylog)
+    lessons = get_assignments(summarylog.user, summarylog)
     notifications = []
     for lesson, contentnode_id in lessons:
         notifications_started = check_and_created_started(lesson, summarylog.user_id, contentnode_id, summarylog.end_timestamp)
@@ -202,7 +202,7 @@ def parse_summarylog(summarylog):
     if summarylog.progress < 1.0:
         return
 
-    lessons = get_assignments(summarylog, summarylog)
+    lessons = get_assignments(summarylog.user, summarylog)
     notifications = []
     for lesson, contentnode_id in lessons:
         # Now let's check completed resources and lessons:
@@ -290,13 +290,16 @@ def parse_attemptslog(attemptlog):
     for the user & resource
     """
     # This Event should not be triggered when a Learner is interacting with an Exercise outside of a Lesson:
-    lessons = get_assignments(attemptlog, attemptlog.masterylog.summarylog, attempt=True)
+    lessons = get_assignments(attemptlog.user, attemptlog.masterylog.summarylog, attempt=True)
     if not lessons:
         return
     # get all the attempt logs on this exercise:
     failed_interactions = []
     attempts = AttemptLog.objects.filter(masterylog_id=attemptlog.masterylog_id)
+
+    # NOTE: saw at elast one error here where failed['correct'] raised a key error
     failed_interactions = [failed for attempt in attempts for failed in attempt.interaction_history if failed['correct'] == 0]
+
     # More than 3 errors in this mastery log:
     needs_help = len(failed_interactions) > 3
 

--- a/kolibri/core/notifications/test/test_api.py
+++ b/kolibri/core/notifications/test/test_api.py
@@ -184,10 +184,10 @@ class NotificationsAPITestCase(APITestCase):
     @patch('kolibri.core.notifications.api.save_notifications')
     def test_parse_examlog(self, save_notifications):
         examlog = ExamLog.objects.create(exam=self.exam, user=self.user1)
-        parse_examlog(examlog)
+        parse_examlog(examlog, local_now())
         assert save_notifications.called is False
         examlog.closed = True
-        parse_examlog(examlog)
+        parse_examlog(examlog, local_now())
         assert save_notifications.called
         notification = save_notifications.call_args[0][0][0]
         assert notification.notification_object == NotificationObjectType.Quiz
@@ -196,7 +196,7 @@ class NotificationsAPITestCase(APITestCase):
     @patch('kolibri.core.notifications.api.save_notifications')
     def test_create_examlog(self, save_notifications):
         examlog = ExamLog.objects.create(exam=self.exam, user=self.user1)
-        create_examlog(examlog)
+        create_examlog(examlog, local_now())
         assert save_notifications.called
         notification = save_notifications.call_args[0][0][0]
         assert notification.notification_object == NotificationObjectType.Quiz
@@ -218,7 +218,17 @@ class NotificationsAPITestCase(APITestCase):
                                                 interaction_history=interactions
                                                 )
         parse_attemptslog(attemptlog1)
-        assert save_notifications.called is False
+        assert save_notifications.called
+        save_notifications.reset_mock()
+        attemptlog2 = AttemptLog.objects.create(masterylog=masterylog, sessionlog=log,
+                                                user=self.user1, start_timestamp=now,
+                                                end_timestamp=now, time_spent=1.0,
+                                                complete=True, correct=1,
+                                                hinted=False, error=False,
+                                                interaction_history=interactions
+                                                )
+        parse_attemptslog(attemptlog2)
+        assert save_notifications.called
         # more than 3 attempts will trigger the help notification
         interactions.append({"type": "answer", "correct": 0})
         attemptlog2 = AttemptLog.objects.create(masterylog=masterylog, sessionlog=log,

